pkg/apis/migration/migration.go

```diff
@@ -92,18 +92,21 @@ const (
 	CRDPlural = "cnsvspherevolumemigrations"
 )
 
-// onceForVolumeMigrationService is used for initializing the VolumeMigrationService singleton.
-var onceForVolumeMigrationService sync.Once
-
-// volumeMigrationInstance is instance of volumeMigration and implements interface for VolumeMigrationService
-var volumeMigrationInstance *volumeMigration
-
-var volumeMigrationServiceInitErr error
+var (
+	// volumeMigrationInstance is instance of volumeMigration and implements interface for VolumeMigrationService
+	volumeMigrationInstance *volumeMigration
+	// volumeMigrationInstanceLock is used for handling race conditions during read, write on volumeMigrationInstance
+	volumeMigrationInstanceLock sync.Mutex
+	// volumeMigrationServiceInitErr is used for capturing the errors while initializing migration service instance
+	volumeMigrationServiceInitErr error
```

Comment:
I dont think this needs to be a global variable anymore. Lets say the first call to `GetVolumeMigrationService` fails and sets `volumeMigrationServiceInitErr`. Even if a subsequent call initializes `volumeMigrationInstance`, we're returning `nil, volumeMigrationServiceInitErr` on line 169

Reply:
Makes sense